frontend/app/src/components/Sidebar/Sidebar.tsx

```diff
@@ -270,121 +233,120 @@ class Sidebar extends PureComponent<SidebarProps, State> {
     return logo
   }
 
-  public render(): ReactNode {
-    const { collapsedSidebar, sidebarWidth, showSidebarCollapse } = this.state
-    const {
-      appPages,
-      chevronDownshift,
-      children,
-      hasElements,
-      onPageChange,
-      currentPageScriptHash,
-      hideSidebarNav,
-      expandSidebarNav,
-      navSections,
-    } = this.props
-
-    const hasPageNavAbove = appPages.length > 1 && !hideSidebarNav
-
-    // Handles checking the URL params
-    const isEmbedded = isEmbed() && !isColoredLineDisplayed()
-    // If header decoration visible, move sidebar down so decoration doesn't go below it
-    const sidebarAdjust = !isEmbedded && this.headerDecorationVisible()
-
-    // The tabindex is required to support scrolling by arrow keys.
-    return (
-      <>
-        <StyledSidebarOpenContainer
-          chevronDownshift={chevronDownshift}
-          data-testid="stSidebarCollapsedControl"
+  const hasPageNavAbove = appPages.length > 1 && !hideSidebarNav
+
+  // Handles checking the URL params
+  const isEmbedded = isEmbed() && !isColoredLineDisplayed()
+  // If header decoration visible, move sidebar down so decoration doesn't go below it
+  const sidebarAdjust = !isEmbedded && headerDecorationVisible()
+
+  function headerDecorationVisible(): boolean {
```

Comment:
it looks like this function does not use any information from the component itself and, thus, can be a standalone function outside of the component render lifecycle